provider/image.go

```diff
@@ -101,16 +104,9 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 	if err != nil {
 		return "", nil, err
 	}
-	cache, err := marshalCachedImages(inputs["build"])
-	if err != nil {
-		return "", nil, err
-	}
-
-	build.CachedImages = cache
```

Comment:
`marshalBuildAndApplyDefaults` now handles marshaling and setting `CachedImages`.